libs/wingc/src/parser.rs

```diff
@@ -1460,11 +1460,8 @@ impl<'s> Parser<'s> {
 			}
 			match interface_element.kind() {
 				"method_signature" => {
-					let method_name = self.node_symbol(&interface_element.child_by_field_name("name").unwrap());
-					let func_sig = self.build_function_signature(&interface_element, phase);
-					match (method_name, func_sig) {
-						(Ok(method_name), Ok(func_sig)) => methods.push((method_name, func_sig)),
-						_ => {}
+					if let Ok((method_name, func_sig)) = self.build_interface_method(interface_element, Phase::Preflight) {
```

Comment:
I suspect a hard coded `Phase::Preflight` here is wrong. What happens if we define an interface in an inflight context. Shouldn't the methods be implicitly inflight:
```wing
let f = inflight () = {
  interface MyIf { ... }; // All methods should be implicitly inflight
  class MyClass implements MyIf { ... }; // Because MyIf can only be used inflight
  let c = new MyClass();
  c.inflight_method();
};
```
If I'm right here, maybe add a test case.

Reply:
Mmm I think you're right.

I reckon it's probably OK to disallow interfaces at the top level, but I'll open a separate issue for it